launch/launch/actions/execute_process.py

```diff
@@ -716,6 +735,31 @@ async def __execute_process(self, context: LaunchContext) -> None:
             self.__logger.error("process has died [pid {}, exit code {}, cmd '{}'].".format(
                 pid, returncode, ' '.join(cmd)
             ))
+            # respawn the process that abnormally died
+            if not context.is_shutdown and self.__respawn:
+                respawn_flag = False
+                if self.__respawn_delay > 0:
+                    try:
+                        # wait for a timeout(`self.__respawn_delay`) to respawn the process
+                        # and handle shutdown event with future(`self.__shutdown_future`)
+                        # to make sure `ros2 launch` exit in time
+                        await asyncio.wait_for(
+                            self.__shutdown_future,
+                            timeout=self.__respawn_delay,
+                            loop=context.asyncio_loop
+                        )
+                    except asyncio.exceptions.TimeoutError:
+                        respawn_flag = True
+                        self.__shutdown_future = create_future(context.asyncio_loop)
+                    except Exception:
+                        self.__logger.error(
+                            'exception occurred while executing asyncio.wait_for:\n'
+                            '{}'.format(traceback.format_exc())
+                        )
+                if self.__respawn_delay == 0 or respawn_flag:
```

Comment:
@iuhilnehc-ynos nit: instead of checking for equality, set the `respawn_flag` to true in an `else` clause if `respawn_delay` is not greater than 0.

Reply:
@hidmic 
Thanks for your guide. I'll update it.